test/gtest/ucp/test_ucp_wireup.cc

```diff
@@ -1228,6 +1228,75 @@ UCP_INSTANTIATE_TEST_CASE_TLS(test_ucp_wireup_unified, rc, "rc")
 UCP_INSTANTIATE_TEST_CASE_TLS(test_ucp_wireup_unified, ud, "ud")
 UCP_INSTANTIATE_TEST_CASE_TLS(test_ucp_wireup_unified, rc_dc, "rc,dc")
 
+class select_ep_transport : public test_ucp_wireup {
+public:
+    static void get_test_variants(std::vector<ucp_test_variant>& variants)
+    {
+        add_variant_with_value(variants, UCP_FEATURE_RMA, TEST_RMA, "rma");
+    }
+
+    char *get_selected_transport(const entity *e)
+    {
+        const ucp_worker *worker    = e->worker();
+        ucp_lane_index_t lane_index = worker->ep_config[0].key.rma_lanes[0];
+        ucp_rsc_index_t rsc_index   =
+                worker->ep_config[0].key.lanes[lane_index].rsc_index;
+        return e->ucph()->tl_rscs[rsc_index].tl_rsc.tl_name;
+    }
+
+    void verify_symmetric_tl_selection(const std::string& num_eps1,
+                                       const std::string& num_eps2,
+                                       const std::string& expected_tl)
+    {
+        /* First context initialized with num_eps above threshold */
+        modify_config("NUM_EPS", num_eps1);
+        entity *e1 = create_entity();
+
+        /* Second context initialized with num_eps below threshold */
+        modify_config("NUM_EPS", num_eps2);
+        entity *e2 = create_entity();
+
+        e1->connect(e2, get_ep_params());
+        char *selected1 = get_selected_transport(e1);
+
+        e2->connect(e1, get_ep_params());
+        char *selected2 = get_selected_transport(e2);
+
+        /* Verify that selection is the same for both eps */
+        ASSERT_STREQ(selected1, selected2);
+
+        /* Verify correct transport was selected */
+        ASSERT_STREQ(selected1, expected_tl.c_str());
```

Comment:
but do we know for sure that the code will crash (and not only print a message and continue)?
Also, for `select_rc_for_32_eps_v2` and `select_rc_for_32_eps_v1` tests it is indeed relevant, because we want to verify RC

Reply:
if there is an error message from the code it will cause gtest to fail